src/app/contexts/EventContext/index.test.jsx

```diff
@@ -0,0 +1,118 @@
+import React, { useContext, useEffect, useState } from 'react';
+import { render } from '@testing-library/react';
+import { func } from 'prop-types';
+import { useWindowEvent } from './useWindowEvent';
+import { useHandlerMap } from './useHandlerMap';
+
+const { EventContextProvider, EventContext } = require('./index');
+
+const Component = () => {
+  useContext(EventContext);
+  return null;
+};
+
+jest.mock('react', () => {
+  const original = jest.requireActual('react');
+  return {
+    ...original,
+    useContext: jest.fn().mockImplementation(original.useContext),
+    useState: jest.fn().mockImplementation(original.useState),
+    useEffect: jest.fn().mockImplementation(original.useEffect),
+  };
+});
+
+jest.mock('./useWindowEvent', () => {
+  const { useWindowEvent: _useWindowEvent } = jest.requireActual(
+    './useWindowEvent',
+  );
+  return {
+    useWindowEvent: jest.fn().mockImplementation(_useWindowEvent),
```

Comment:
If you want to check if a function is called but keep the original functionality then I think this is what spies are for e.g. `jest.spyOn` see this commit I made https://github.com/bbc/simorgh/pull/3993/commits/341ef49ceda4496165c231a77065973b25dfdd53

Reply:
Cool! I've merged the branch in. Thanks!